persistence.js

```diff
@@ -451,30 +455,18 @@ class LevelPersistence extends EventEmitter {
   }
 }
 
-function withClientId (sub) {
-  return {
-    topic: sub.topic,
-    clientId: this.id,
-    qos: sub.qos
-  }
-}
-
-function toSubKey (sub) {
-  return `${subByClientKey(sub.clientId)}:${sub.topic}`
-}
-
-function addSubToTrie (sub) {
-  let add
-  const matched = this.match(sub.topic)
+function addSubToTrie (trie, sub) {
+  let add = false
+  const matched = trie.match(sub.topic)
   if (matched.length > 0) {
     add = true
-    for (let i = 0; i < matched.length; i++) {
-      if (matched[i].clientId === sub.clientId) {
-        if (matched[i].qos === sub.qos) {
+    for (const match of matched) {
```

Comment:
for..i loops are muchg more performant then forof loops

Reply:
They used to be, but that is already some time ago ;-)
> So my conclusion is simple, it doesn’t seem to matter what loop you choose, they all perform equally enough you should care about readability of the code, and not performance of a loop.
https://javascript.plainenglish.io/the-only-javascript-loop-benchmark-that-matters-for-the-most-of-us-77ec819eb23e

Even Qlobber uses `for..of` these days ;-)
https://github.com/davedoesdev/qlobber/blob/814dae115b00aa38a56ff771142c418555014cb6/aedes/qlobber-sub.js#L36

Reply:
Ok good so :)